### PR TITLE
Minor optimization on OnReceive method of RemoteNodes 

### PR DESCRIPTION
**Pull request by vncoelho**
Draft made on the top of #807, without conflicts.

This PR adds a new protected variable in the Connection.cs class, which is used for avoiding the switch of the RemoteNode when Connection processed basic TCP messages.

### CODE REVIEW

**Comment by vncoelho** [comment]
@igormcoelho is worried about the asynchronous behavior.

File: neo/Network/P2P/RemoteNode.cs [hunk 1]
```diff
@@ -120,30 +120,31 @@ protected override void OnData(ByteString data)
...
-                    OnVerack();
-                    break;
-                case ProtocolHandler.SetFilter setFilter:
-                    OnSetFilter(setFilter.Filter);
-                    break;
-                case PingPayload payload:
-                    OnPingPayload(payload);
-                    break;
-            }
+            if(!base.messageProcessed)
```
Comment (erikzhang) [inline neo/Network/P2P/RemoteNode.cs]: I think the performance of this `if` is the same as the `switch`. There is almost no difference.

Comment (vncoelho) [reply neo/Network/P2P/RemoteNode.cs]: Igor told me the same....ehauwhahau

Comment (vncoelho) [reply neo/Network/P2P/RemoteNode.cs]: @igormcoelho, please help with the final decision.

Comment (vncoelho) [reply neo/Network/P2P/RemoteNode.cs]: @erikzhang, in terms of threads and assync acess, is there any issue?

Comment (erikzhang) [reply neo/Network/P2P/RemoteNode.cs]: There is no assync access in actors.

Comment (igormcoelho) [reply neo/Network/P2P/RemoteNode.cs]: I'll quote Donald Knuth:

“The real problem is that programmers have spent far too much time worrying about efficiency in the wrong places and at the wrong times; premature optimization is the root of all evil (or at least most of it) in programming.”

Leave the switch, its much more readable and pehaps much better than ifs.

**Pull request closed without merging**